### PR TITLE
fix build for tinywl

### PR DESCRIPTION
**Pull request by wineee**

### CODE REVIEW

File: examples/tinywl/main.cpp [hunk 1]
```diff
@@ -24,10 +24,14 @@
 #include <qwinputdevice.h>
 #include <qwkeyboard.h>
 #include <qwpointer.h>
-#include <qwbackendinterface.h>
 #include <qwtexture.h>
 
-/*class TinywlServer : public QObject
+#define WLR_USE_UNSTABLE
+extern "C" {
+#include <wlr/util/log.h>
```
Comment (zccrs) [inline examples/tinywl/main.cpp:31]: why not use `qwlogging.h`

File: examples/tinywl/main.cpp [hunk 2]
```diff
@@ -519,7 +532,10 @@ void TinywlServer::processCursorMotion(uint32_t time)
         if (maxSize.height() == 0)
             maxSize.setHeight(99999);
 
-        auto currentGeoBox = grabbedView->xdgToplevel->getGeometry();
+        wlr_box current_geo_box;
+        qw_xdg_surface::from(grabbedView->xdgToplevel->handle()->base)->get_geometry(&current_geo_box);
+        auto currentGeoBox = QRect(current_geo_box.x, current_geo_box.y, current_geo_box.width, current_geo_box.height);
```
Comment (zccrs) [inline examples/tinywl/main.cpp:537]: Maybe needs a qw_box

File: examples/tinywl/CMakeLists.txt [hunk 3]
```diff
@@ -16,6 +16,8 @@ if(PROJECT_IS_TOP_LEVEL) # if build as a separated project
     else()
         find_package(QT NAMES Qt6 REQUIRED COMPONENTS Core)
     endif()
+else()
+    ws_generate(server wayland-protocols stable/xdg-shell/xdg-shell.xml xdg-shell-protocol)
```
Comment (zccrs) [inline examples/tinywl/CMakeLists.txt:20]: why needs generate in examples?

**Review by zccrs — CHANGES_REQUESTED** [review]

File: examples/tinywl/CMakeLists.txt (hunk 3 above)
Comment (wineee) [reply examples/tinywl/CMakeLists.txt:20]: CMAKE_CURRENT_BINARY_DIR are different for qwlroots and tinywl

**Pull request merged**